### PR TITLE
update to OpenRefine 3.1 and some basic enhancements by psychemedia

### PR DESCRIPTION
**Pull request by felixlohmeier**
I would like to suggest some improvements from @psychemedia (and some minor additions by myself) for the main repo:
* update to OpenRefine 3.1 (current stable version)
* set OpenRefine workspace dir relative to $HOME
* set OpenRefine port to 3333
* add python client for OpenRefine in requirements.txt
* remove tar file after extracting
* explain different launch options in README.md

There is also a demo notebook in https://github.com/ouseful-PR/openrefineder. It works with the suggested changes in this pull request but I think that it is still work in progress.

### CODE REVIEW

**Comment by betatim** [comment]
Looks good to me. Thanks for refurbishing this!

File: nbopenrefineproxy/handlers.py [hunk 1]
```diff
@@ -33,6 +36,6 @@ def get(self, *args):
 def setup_handlers(web_app):
     web_app.add_handlers('.*', [
         (ujoin(web_app.settings['base_url'], 'openrefine/(.*)'),
-         OpenRefineProxyHandler, dict(state={})),
+         OpenRefineProxyHandler, dict(state={'port':3333})),
```
Comment (betatim) [inline nbopenrefineproxy/handlers.py:39]: What is the advantage of doing this?

**Comment by betatim** [comment]
I'll merge this now and if we need to we can make a new PR to change the port behaviour again.

File: nbopenrefineproxy/handlers.py (hunk 1 above)
Comment (felixlohmeier) [reply nbopenrefineproxy/handlers.py:39]: Thanks for merging!

3333 is the default port used by OpenRefine. Specifying it here makes it easier to connect with the python client (which tries to reach the OpenRefine server on 127.0.0.1:3333 by default). @psychemedia suggested this in https://github.com/betatim/openrefineder/issues/6#issuecomment-449406846

Comment (betatim) [reply nbopenrefineproxy/handlers.py:39]: Thanks. Makes sense!

**Pull request merged**